frontend/front/src/pages/Surveyor/dashboard/Dashboard.js

```diff
@@ -7,39 +7,8 @@ const Dashboard = () => {
   const { data, isLoading, isSuccess, isError, error } =
     useGetSurveyorAssignmentQuery();
 
-  let content = "no data";
-
-  if (isLoading) {
-    content = <Box> Is loading</Box>;
-  } else if (isSuccess) {
-    content = <Box>Status: Is successful</Box>;
-    console.log("dashboard", data);
-    // content = data.map((item) => {
-    //   return (
-    //     <Box key={item.id} border={1}>
-    //       <HomeItem data={item} />
-    //       <Button
-    //         onClick={() =>
-    //           updateHomeData({ ...item, completed: !item.completed })
-    //         }
-    //       >
-    //         Toggle Completed
-    //       </Button>
-    //       <Button onClick={() => deleteHomeData(item)}>delete</Button>
-    //     </Box>
-    //   );
-    // });
-  } else if (isError) {
-    content = <Box>{error.error}</Box>;
-  }
-
   return (
     <Box>
-      <Box m={3} p={2}>
-        <Typography>Json Server api</Typography>
-        {content}
-      </Box>
-
       <Box display={"flex"} justifyContent="center">
         <ListView />
```

Comment:
My suggestion is that we should still require this verification. Maybe we could refactor the code like this:

`{isLoading ? (<Loader />) : isError ? (<Box>{error.error}</Box>) : (<ListView />)}`

In the src > components directory, we could create a Loader.js file with the following code:

```
import React from "react";
import { styled } from "@mui/material/styles";
import { CircularProgress } from "@mui/material";

const LoaderContainer = styled("div")({
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
  height: "100vh",
});

function Loader() {
  return (
    <LoaderContainer>
      <CircularProgress />
    </LoaderContainer>
  );
}

export default Loader;

```

Hey @mattdelsordo , let me know what you think about it?

Reply:
Oh, I didn't realize this was hooked in to the same `useGetSurveyorAssignmentQuery()`. I agree that there should be some sort of loader + error handling. What I've been doing on other pages is something like
```
isLoading ? <CircularProgress /> : <ListView />
```
to show the progress animation instead of the main page content (so basically exactly what you have there) but then using [error snackbars](https://mui.com/material-ui/react-snackbar/) for error messages.

Reply:
@mattdelsordo @thiagobardini Thanks for your input guys! Please let me know how you would like to proceed with this and I can move forward with making the necessary changes :)

Reply:
I'd say start with my suggestion and we can go from there.

Reply:
Sounds good!

Reply:
if you don't mind, could you create a Loader component so that we can replicate the same loading behavior on the other pages?